### ISSUE TITLE
Color contrast: visually hidden file input should not go into needs review

### ISSUE THREAD

**Issue by marcysutton** [dequelabs/axe-core#622]
In [Sarah Drasner's alt text generator Codepen](https://s.codepen.io/sdras/debug/9b9d4ab09d8a79bf295b07044b268f39), the visually hidden file input is going into needs review with the generic message `Unable to determine contrast ratio`.

![visually hidden input open in Chrome devtools](https://user-images.githubusercontent.com/1045233/33225600-6082f7f4-d12f-11e7-85c1-429890a6c7c9.png)

The label shows a focus state when the user tabs onto the input, which is hidden from view with this CSS:
```
.inputfile {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    overflow: hidden;
    position: absolute;
    z-index: -1;
}
.inputfile:focus + label {
    outline: 1px dotted #000;
    outline: -webkit-focus-ring-color auto 5px;
}
```
The important bit in there for contrast purposes is `opacity:0`. If an element is invisible, we shouldn't put it into needs review for color contrast. We can expand `axe.commons.dom.isVisible` to check for opacity:0, and then exclude it from the color-contrast-matches. Would that cause any issues? I've already got a fix for it, going to submit it as a separate PR.

https://s.codepen.io/sdras/debug/9b9d4ab09d8a79bf295b07044b268f39

**Comment by marcysutton** [comment]
To capture some of the offline conversation:
> This isn't a false positive. It's highly specific, so we'll probably never see a case like it again, it adds complexity to the most difficult to maintain part of our code base, and it's actually a side-effect of a real WCAG violation (3.2.2, on input, e.g., give users a chance to review before submitting stuff)

I do think we'll see this again, and it's pretty simple code. But given the on input issue I'm going to close this one.
